### PR TITLE
add service published event type

### PR DESCRIPTION
**Pull request by ishankhare07**
also added ksvc specific fields to extensions

Signed-off-by: Ishan Khare <me@ishankhare.dev>

### CODE REVIEW

**Review by afrittoli — COMMENTED** [review]
Thank you @ishankhare07 for your PR.

Service created and published event types are not in the spec today, and we should not add them to the SDK until they are.

IIUC, this is a follow-up to https://github.com/cdfoundation/sig-events/issues/107, which is great, but we should first have an agreement on the events before we add this to the SDK.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,26 @@
-.idea/
```
[the remark this answers is not in the document]
Comment (ishankhare07) [reply .gitignore:1]: I think I removed is accidentally, let me fix that

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,26 @@
...
-cde/sdk/go/build/*
+
+# Created by https://www.toptal.com/developers/gitignore/api/vim
+# Edit at https://www.toptal.com/developers/gitignore?templates=vim
```
[the remark this answers is not in the document]
Comment (ishankhare07) [reply .gitignore:3]: yes sure, they were generated by the upstream helper service and don't need to be there.

**Review by afrittoli — COMMENTED** [review]
Thank you @ishankhare07 for the updates!

The SDK is still in this repo - as discussed, but the spec has moved over to github.com/cdevents/spec. Could you make the spec part of the PR there please?

I should put a notice here in the readme that redirects to the new location, sorry about that.

**Comment by ishankhare07** [comment]
Hey @afrittoli just checked that, it seems like github.com/cdevents/spec documents already have the new nomenclature in place - referring https://github.com/cdevents/spec/blob/main/continuous-deployment-pipeline-events.md#continuous-deployment-events 

So I think a PR there won't be required. Regarding the changes in this repo, since we've already moved the spec from here I think we can remove those files in future when we plan on doing so and keep it as it is.

However if you want I can discard the spec related changes from this PR and only keep the SDK changes.

Let me know what you think.

**Comment by afrittoli** [comment]
> Hey @afrittoli just checked that, it seems like github.com/cdevents/spec documents already have the new nomenclature in place - referring https://github.com/cdevents/spec/blob/main/continuous-deployment-pipeline-events.md#continuous-deployment-events

Yes, the new spec uses the new nomenclature, I updated it myself. 
It does not include the new event that you're proposing, which is what I was referring to.

> 
> So I think a PR there won't be required. 

You'll still need to add the `service.published` event?

> Regarding the changes in this repo, since we've already moved the spec from here I think we can remove those files in future when we plan on doing so and keep it as it is.

Yes, I will update this repo and remove those files.

> 
> However if you want I can discard the spec related changes from this PR and only keep the SDK changes.

If you don't mind? I'd rather keep a clean git history 🙏 

> 
> Let me know what you think.

**Comment by ishankhare07** [comment]
> You'll still need to add the service.published event
Got it, I'll open a PR for this on the spec repo.

**Comment by m-linner-ericsson** [comment]
Should we close this PR if we have one in the spec repo?

**Comment by afrittoli** [comment]
@ishankhare07 Closing this in favour of https://github.com/cdevents/spec/pull/31 - feel free to re-open if needed.

**Pull request closed without merging**